### PR TITLE
[eeprom_tlv_info] Optimize EEPROM data process by using visitor pattern

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Description

1. Use visitor pattern accessing EEPROM data
2. Provide utility functions to access redis data base

#### Motivation and Context

Remove EEPROM cache file and use DB instead

#### How Has This Been Tested?

Manual test and regression test

### CODE REVIEW

File: sonic_platform_base/sonic_eeprom/eeprom_tlvinfo.py [hunk 1]
```diff
@@ -695,3 +615,165 @@ def checksum_field_size(self):
...
+               e[tlv_index] == self._TLV_CODE_CRC_32:
+                break
+            tlv_index += e[tlv_index+1] + 2
+        visitor.visit_end(e)
+
+
+class EepromDefaultVisitor:
+    def visit_header(self, eeprom_id, version, header_length):
+        """Visit EEPROM header. If EEPROM header is not present, this method is still called, 
+           but the inpurt arguments will all be None.
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_tlvinfo.py]: s/inpurt/input/

**Review by jleveque — CHANGES_REQUESTED** [review]

File: sonic_platform_base/sonic_eeprom/eeprom_base.py [hunk 2]
```diff
@@ -27,6 +27,8 @@ def __init__(self, path, format, start, status, readonly):
         self.s = start
         self.u = status
         self.r = readonly
+        # Warning: the following members are deprecated, the parsed EEPROM data is stored in the
+        # Redis STATE_DB, cached data should be feched from STATE_DB.EEPROM_INFO.
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_base.py]: s/feched/fetched/

File: sonic_platform_base/sonic_eeprom/eeprom_base.py [hunk 3]
```diff
@@ -47,6 +49,9 @@ def check_status(self):
             return 'ok'
 
     def set_cache_name(self, name):
+        # Warning: this method is deprecated, the parsed EEPROM data is stored in the
+        # Redis STATE_DB, cached data should be feched from STATE_DB.EEPROM_INFO.
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_base.py]: s/feched/fetched/

File: sonic_platform_base/sonic_eeprom/eeprom_base.py [hunk 4]
```diff
@@ -214,6 +219,9 @@ def open_eeprom(self):
         using_eeprom = True
         eeprom_file = self.p
         try:
+        # Warning: cache file is deprecated, the parsed EEPROM data is stored in the
+        # Redis STATE_DB, cached data should be feched from STATE_DB.EEPROM_INFO. This
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_base.py]: s/feched/fetched/

File: sonic_platform_base/sonic_eeprom/eeprom_base.py [hunk 5]
```diff
@@ -240,6 +248,9 @@ def read_eeprom_bytes(self, byteCount, offset=0):
             # expect, the file may be corrupt. Delete it and try again, this
             # time reading from the actual EEPROM.
             if len(o) != byteCount and not self.cache_update_needed:
+                # Warning: cache file is deprecated, the parsed EEPROM data is stored in the
+                # Redis STATE_DB, cached data should be feched from STATE_DB.EEPROM_INFO. This
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_base.py]: s/feched/fetched/

File: sonic_platform_base/sonic_eeprom/eeprom_base.py [hunk 6]
```diff
@@ -277,6 +288,9 @@ def write_eeprom(self, e):
         self.write_cache(e)
 
     def write_cache(self, e):
+        # Warning: this method is deprecated, the parsed EEPROM data is stored in the
+        # Redis STATE_DB, cached data should be feched from STATE_DB.EEPROM_INFO.
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_base.py]: s/feched/fetched/

File: sonic_platform_base/sonic_eeprom/eeprom_base.py [hunk 7]
```diff
@@ -290,6 +304,9 @@ def write_cache(self, e):
                     F.close()
 
     def update_cache(self, e):
+        # Warning: this method is deprecated, the parsed EEPROM data is stored in the
+        # Redis STATE_DB, cached data should be feched from STATE_DB.EEPROM_INFO.
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_base.py]: s/feched/fetched/

File: sonic_platform_base/sonic_eeprom/eeprom_base.py [hunk 8]
```diff
@@ -240,6 +248,9 @@ def read_eeprom_bytes(self, byteCount, offset=0):
             # expect, the file may be corrupt. Delete it and try again, this
             # time reading from the actual EEPROM.
             if len(o) != byteCount and not self.cache_update_needed:
+                # Warning: cache file is deprecated, the parsed EEPROM data is stored in the
+                # Redis STATE_DB, cached data should be feched from STATE_DB.EEPROM_INFO. This
+                # code need to be adjusted once cache file is completely removing from the system.
```
Comment (jleveque) [inline sonic_platform_base/sonic_eeprom/eeprom_base.py]: s/need/needs/
s/removing/removed/

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by keboliu — APPROVED** [review]

**Pull request closed without merging**